cmd/ooohh-api/main.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/boltdb/bolt"
+	"github.com/pkg/errors"
+
+	"github.com/dlmiddlecote/ooohh"
+	"github.com/dlmiddlecote/ooohh/pkg/service"
+)
+
+func main() {
+	if err := run(); err != nil {
+		fmt.Fprintf(os.Stdout, "error: %v", err)
+		os.Exit(1)
+	}
+}
+
+func run() error {
+
+	var err error
+	var s ooohh.Service
+	{
+		db, err := bolt.Open("/tmp/bolt.db", 0600, nil)
+		if err != nil {
+			return errors.Wrap(err, "opening db")
+		}
+		defer db.Close()
+	
+		now := func() time.Time {
+			return time.Now()
+		}
+
+		s, err = service.NewService(db, now)
+	}
```

Comment:
Out of interest, what's the reason for nesting this code block? Is it for the benefit of the `defer`? E.g. it will run db.Close() after `service.NewService(...)`?

Reply:
It’s mainly stylistic. The block creates a new scope, so anything declared inside the block won’t be available outside of it, unless it was declared outside (like the ooohh.Service). It’s just to group the component initialisation together. The defer is unaffected, and will be called after “run” exits.